cpp/src/NVStrings_array.cu

```diff
@@ -105,6 +105,33 @@ NVStrings* NVStrings::gather( const int* pos, unsigned int elems, bool bdevmem )
     return rtn;
 }
 
+// create a new instance containing only the strings where the corresponding mask value is true
+NVStrings* NVStrings::gather( const bool* mask, bool bdevmem )
```

Comment:
So the main issue with this is that in libcudf, we don't have columns of type `bool`, we have columns of type `cudf::bool8` (which is just a wrapper for `char`).

See https://github.com/rapidsai/cudf/blob/branch-0.8/cpp/src/utilities/wrapper_types.hpp#L346

There's no guarantee that `sizeof(bool) == sizeof(cudf::bool8)`. So it's unsafe to simply just cast a `cudf::bool8` column to a `bool` column.

If this function meets an immediate need, then feel free to move forward with it. However, in the medium term as we merge codebases, functions like this should accept a `cudf::bool8` mask instead of a `bool` mask.

Reply:
Merging codebases has not yet started/scheduled and likely take awhile. I don't want to hold up this request until then.